network/datadog_checks/network/check_windows.py

```diff
@@ -74,6 +103,57 @@ def _cx_counters_psutil(self, tags=None):
             }
             self.submit_devicemetrics(iface, metrics, tags)
 
+    def _get_tcp_stats(self, inet):
+        stats = TCPSTATS()
+        try:
+            Iphlpapi.GetTcpStatisticsEx(byref(stats), inet)
+        except OSError as e:
+            self.log.error("OSError: %s", e)
+            return None
+        return stats
+
+    def _tcp_stats(self, tags):
+        """
+        Collect metrics from Microsoft's TCPSTATS
+        """
+        tags = [] if tags is None else tags
+
+        tcpstats_dict = {
+            'dwActiveOpens': '.active_opens',
+            'dwPassiveOpens': '.passive_opens',
+            'dwAttemptFails': '.attempt_fails',
+            'dwEstabResets': '.established_resets',
+            'dwCurrEstab': '.current_established',
+            'dwInSegs': '.in_segs',
+            'dwOutSegs': '.out_segs',
+            'dwRetransSegs': '.retrans_segs',
+            'dwInErrs': '.in_errors',
+            'dwOutRsts': '.out_resets',
+            'dwNumConns': '.connections',
+        }
+
+        proto_dict = {}
+        tcp4stats = self._get_tcp_stats(socket.AF_INET)
+        if tcp4stats:
+            proto_dict["tcp4"] = tcp4stats
+        tcp6stats = self._get_tcp_stats(socket.AF_INET6)
+        if tcp6stats:
+            proto_dict["tcp6"] = tcp6stats
+
+        tcpAllstats = TCPSTATS()
+        # Create tcp metrics that are a sum of tcp4 and tcp6 metrics
+        if 'tcp4' in proto_dict and 'tcp6' in proto_dict:
+            for fieldname, _ in tcpAllstats._fields_:
+                tcp_sum = getattr(proto_dict['tcp4'], fieldname) + getattr(proto_dict['tcp6'], fieldname)
+                setattr(tcpAllstats, fieldname, tcp_sum)
+            proto_dict["tcp"] = tcpAllstats
+
+        for proto, stats in proto_dict.items():
+            for fieldname in tcpstats_dict:
+                fieldvalue = getattr(stats, fieldname)
+                metric_name = "system.net." + str(proto) + tcpstats_dict[fieldname]
+                self.submit_netmetric(metric_name, fieldvalue, tags)
+
     def _parse_protocol_psutil(self, conn):
```

Comment:
Is this what Linux implementation has. Metrics for tcp4, tcp6 and just tcp?

Reply:
The linux implementation only submits the combined value for these metrics. The tcp4 and tcp6 metrics added by this change are not available on the linux implementation